### DISCUSSION TITLE
Convert back from a dense batch to a sparse batch

### DISCUSSION THREAD

**Discussion by AnimeshSinha1309** [pyg-team/pytorch_geometric#4540] [Q&A]
I need to use the `pyg.utils.to_dense_batch` function for getting `dense_mincut_pooling` to work.
Now, for the future layers, the tutorial on `dense_mincut_pooling` shows that we need to use `DenseGraphConv` layers.
However, there are some custom layers that I have implemented which do not have their Dense versions.
So is there some way, even if it is computationally inefficient, to undo the `to_dense_batch` function and get a sparse batch back.
I can't find a function called `to_sparse_batch`, is that by design?

**Comment by rusty1s** [comment]
You should be able to do
```python
x = x.view(-1, x.size(-1)
edge_index, edge_weight = torch_geometric.utils.dense_to_sparse(adj)
```
As you said, this will return a lot of edges (due to how these dense pooling approaches work) which may be utterly inefficient. You may be able to reduce the number of edges by filtering them out based on `edge_weight`:
```python
mask = edge_weight > 0.2
edge_index, edge_weight = edge_index[:, mask], edge_weight[mask]
```
